pkg/systemd/reporting.jsx

```diff
@@ -307,184 +319,124 @@ class BusWorkflowRow extends React.Component {
-            <Button key={"cancel_" + props.label}
+            <Button key={"cancel_" + label}
                     variant="secondary"
-                    onClick={props.onCancelButtonClick}>
+                    onClick={onCancelButtonClick}>
                 {_("Cancel")}
             </Button>
         );
     } else {
         button = (
-            <Button key={"report_" + props.label}
+            <Button key={"report_" + label}
                     variant="primary"
-                    isDisabled={props.problemState !== ProblemState.REPORTABLE}
-                    onClick={props.problemState === ProblemState.REPORTABLE ? props.onReportButtonClick : undefined}>
+                    isDisabled={problemState !== ProblemState.REPORTABLE}
+                    onClick={problemState === ProblemState.REPORTABLE ? onReportButtonClick : undefined}>
                 {_("Report")}
             </Button>
         );
     }
 
     return (
         <Split hasGutter>
-            <SplitItem>{props.label}</SplitItem>
+            <SplitItem>{label}</SplitItem>
             <SplitItem isFilled>
-                {props.problemState === ProblemState.REPORTING && <Spinner size="md" /> }
+                {problemState === ProblemState.REPORTING && <Spinner size="md" /> }
                 {status}
             </SplitItem>
             <SplitItem>{button}</SplitItem>
         </Split>
     );
-}
+};
 
 const reportd_client = cockpit.dbus("org.freedesktop.reportd", { superuser: "try" });
 
-export class ReportingTable extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            workflows: [],
-        };
+export const ReportingTable = ({ problem }) => {
+    const [workflows, setWorkflows] = React.useState([]);
 
+    useInit(() => {
         reportd_client
                 .wait()
-                .then(() => this.getWorkflows(reportd_client),
+                .then(() => getWorkflows(reportd_client),
                       exception => console.error(cockpit.format("Channel for reportd D-Bus client closed: $0", exception.problem || exception.message)));
-    }
+    });
 
-    getWorkflows(client) {
-        client.call("/org/freedesktop/reportd/Service", "org.freedesktop.reportd.Service", "GetWorkflows", [this.props.problem.path])
-                .then((args, options) => this.setState({ workflows: args[0] }),
-                      exception => console.error(cockpit.format("Failed to get workflows for problem $0: $1", this.props.problem.path, (exception.problem || exception.message))));
-    }
+    const getWorkflows = client => {
+        client.call("/org/freedesktop/reportd/Service", "org.freedesktop.reportd.Service", "GetWorkflows", [problem.path])
+                .then((args, _options) => setWorkflows(args[0]),
+                      exception => console.error(cockpit.format("Failed to get workflows for problem $0: $1", problem.path, (exception.problem || exception.message))));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20230927-102042-c578ad22-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)